src/main/service/ResourceCleanupService.ts

```diff
@@ -0,0 +1,59 @@
+import { AWSError, S3 } from 'aws-sdk';
+import {
+  DeleteObjectOutput,
+  DeleteObjectRequest,
+  DeleteObjectsOutput,
+  DeleteObjectsRequest,
+  ObjectIdentifierList,
+} from 'aws-sdk/clients/s3';
+import { PromiseResult } from 'aws-sdk/lib/request';
+
+export const S3_BUCKET = 'promopal';
+
+/**
+ * Generic class used to capture external resource cleanup such as deleting files from S3
+ * */
+export class ResourceCleanupService {
+  private s3: S3;
+
+  constructor(s3: S3) {
+    this.s3 = s3;
+  }
+
+  /**
+   * Cleanup any external resources associated with a promotion
+   * * E.g. Delete S3 images associated with the promotion
+   * @param promotionId the id of the promotion to clean up resources for
+   * */
+  cleanupResourceForPromotion(
+    promotionId: string
+  ): Promise<PromiseResult<DeleteObjectOutput, AWSError>> {
+    const deleteObjectRequest: DeleteObjectRequest = {
+      Bucket: S3_BUCKET,
+      Key: promotionId,
+    };
+
+    return this.s3.deleteObject(deleteObjectRequest).promise();
+  }
+
+  /**
+   * Cleanup any external resources associated with multiple promotions
+   * * E.g. Delete S3 images associated with many promotions
+   * @param promotionIds the id of the promotions to clean up resources for
+   * */
+  cleanupResourceForPromotions(
+    promotionIds: string[]
+  ): Promise<PromiseResult<DeleteObjectsOutput, AWSError>> {
```

Comment:
Just curious 😄  but how does it work if there's an AWSError?

Reply:
Good Q! not sure, I will need to look into this hehe

Reply:
Okie after looking at it, I think it works like this: 

`<PromiseResult<DeleteObjectsOutput, AWSError>` is defined as this:
![image](https://user-images.githubusercontent.com/49849754/113370247-373a5500-9318-11eb-9c72-b2c0dab27a96.png)
So actually, we are returned
`DeleteObjectsOutput & {$response: Response<DeleteObjectsOutput. AWSError>}`

Then we can go into the individual type definitions of `DeleteObjectsOutput, Response, AWSError` to see how the data will be handed to us.

Reply:
Cool!! Thanks for looking into it 👍🏻  Should we check into this object wherever we're making the call to the function?

Reply:
Yeahh I was thinking about that, usually when I see an error its thrown, I haven't seen an instance of `AWSError`. I don't think we need to worry about it, our use case of this is very small just deleting an object regardless if it exists or not.